### PR TITLE
Added top-level comaprator script

### PR DESCRIPTION
**Pull request by judovana**
```
# This is top level script for comparing two jdks.  It 
# This script expects exactly two args - tag:os.arch/dir/archive, two times
# eg: compare_builds.sh  /usr/lib/jvm/java-21-openjdk /my/home/my_custom_tarball.tar.xz
# eg: compare_builds.sh  /my/custom/dir/with_jdk jdk-21.0.4+7:windows.x64
# if the parameter is tag, the temurin jdk of given tag:os.arch is downloaded
# yo can omit the :os.arch part, then it will be guessed... but you may know better
# It copies and creates all in CWD!!!!
# Pass the files/dirs as absolute values, it should not be that hard
# It is useful to have JAVA_HOME pointing to some other JDK used to compile and run supporting files
# otherwise one of the provided JDKs will be duplicated, and used
# for your own safety, do not run it as root
#
# The JDK you are comparing is launched to get various info about itself
# If you need to compare jdks on different platform then they are for, you need to go manually.
```

### CODE REVIEW

**Comment by judovana** [comment]
I iwll add license header and do the linter exercise once we agree it is desirbale work to go in. TY!

File: tooling/compare-builds.sh [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+    export CLASSPATH="${ftureDir}"
+    if uname | grep CYGWIN ; then
+      export CLASSPATH=$(cygpath -m "${CLASSPATH}")
+    fi
+    getKeysForComparablePatch ${WORKDIR}/${jdkName}
+    # we cannot use the JDK we are currently processing. It is broken by the patching
+    PATH="${JAVA_HOME}/bin:${PATH}" bash ./comparable_patch.sh --jdk-dir "${WORKDIR}/${jdkName}" --version-string "$JAVA_VENDOR_VERSION" --vendor-name "$JAVA_VENDOR" --vendor_url "$JAVA_VENDOR_URL" --vendor-bug-url "$JAVA_VENDOR_URL_BUG" --vendor-vm-bug-url "$JAVA_VENDOR_URL_BUG"  2>&1 | tee -a "$LOG"
+  done
+  if [ "x${DO_BREAK:-}" == "xtrue" ] ; then
+    echo "dead" > "${WORKDIR}/${JDK_INFO["first_name"]}/bin/java"
```
[the remark this answers is not in the document]
Comment (judovana) [reply tooling/compare-builds.sh:457]: The `${DO_BREAK:-}"` block  is serving to tempt your toolchain a bit. In the moment  I saw it claims some buiilds to be reproducible., I wanted to modify some parts to see if results are as they should be.  At the end I left only this two simple modification as placeholder where to do something like that  - or use  as it is.  I do not have hard feelings to remove it, but it was useful.

**Comment by karianna** [comment]
linter is upset :-)

**Comment by judovana** [comment]
> linter is upset :-)

Linter is never upset, it just gently reminds!

**Review by andrew-m-leonard — CHANGES_REQUESTED** [review]
I'm not sure at the moment we want to add this as another top-level script, which may just complicate things...

**Comment by judovana** [comment]
Sure. I have no objections to  not  include this one. But word "another" is interesting.  I have not noticed anythig similar around. . I  definitely do not want myself, nor anybody else to  to keep "supporting" two top level wrappers,  but would like to have one which do its job properly.   So I would like to adjust the already existing to do what I need,  or move workloads which are using the current one to this new one.  Can you please pin point me?

**Comment by andrew-m-leonard** [comment]
> Sure. I have no objections to not include this one. But word "another" is interesting. I have not noticed anythig similar around. . I definitely do not want myself, nor anybody else to to keep "supporting" two top level wrappers, but would like to have one which do its job properly. So I would like to adjust the already existing to do what I need, or move workloads which are using the current one to this new one. Can you please pin point me?

So "another" is actually relative to the "top level use case". We already have in this repo the new AQA rebuild tests here for example: https://github.com/adoptium/temurin-build/blob/5035fa3b6bfc1787c94d7b1e91e2979619b9e02a/test/system/reproducibleCompare/playlist.xml#L22

**Comment by smlambert** [comment]
I will close this PR as unnecessary/duplicate, especially as we prefer to use our existing approach and leverage existing AQAvit tools such as TRSS when running and monitoring comparisons and reproducible tests. 

Also as a reminder, PRs should be related to existing issues which are either in backlog or accepted into a quarterly plan by component owners / PMC / project leads, so we can ensure that we are prioritizing our time and resources well.

**Comment by judovana** [comment]
> > Sure. I have no objections to not include this one. But word "another" is interesting. I have not noticed anythig similar around. . I definitely do not want myself, nor anybody else to to keep "supporting" two top level wrappers, but would like to have one which do its job properly. So I would like to adjust the already existing to do what I need, or move workloads which are using the current one to this new one. Can you please pin point me?
> 
> So "another" is actually relative to the "top level use case". We already have in this repo the new AQA rebuild tests here for example:
> 
> https://github.com/adoptium/temurin-build/blob/5035fa3b6bfc1787c94d7b1e91e2979619b9e02a/test/system/reproducibleCompare/playlist.xml#L22

Yah, I have seen that playlist. It did not seems to be doing what I wanted.  I will try to look again.

> AQAvit tools such as TRSS when running and monitoring comparisons and reproducible tests.

Sure, but there is absiolutely no reason an testsuite could generate  more types of results.

**Comment by judovana** [comment]
@smlambert  Actually one more thing.  Three weeks ago I wanted to run compare of one jdk against existing temurins. There was (and afaik still is)  nothing which is helping to glue all the parts together to compare existing build against  released temurins.  Especially under cygwin.  
So I wrote it, and it was well standalone  be well reusable,  tested in cygwin and fedora/rhel.  So I have naturally offered it back to the project from which  the individual parts, and a pretty long readme (which do not mention any top level wrapper) come from (I have searched for that, and found the playlists, and they do not supply what I needed).  So I'm wondering, why it is so hard  for you to be at least semi-welcoming, and  jsut  write. "sorry this is duplicated with already existing toolchain". And even better, to enhance readme how to use that if it is alrerady existing.   Also I'm wondering why one needs a PMC/governing board/whatever agreement, to contribute already existing, simple script.  Which dont take any time of your people.  Except review which would not be that hard considering it is jsut top level glue only. And if so, then it can always go on hold.

**Pull request closed without merging**